### ISSUE TITLE
Improve the Arabic support

### ISSUE THREAD

**Issue by rhdunn** [espeak-ng/espeak-ng#186]
The Arabic support as written needs improving. Specifically:

- [ ] Use more standard Arabic phoneme names (https://en.wikipedia.org/wiki/Arabic_phonology), e.g. `a:` instead of `aa`.
- [ ] Improve/rewrite the rules file so that it has proper Arabic pronunciation rules.
- [ ] Remove as much of the `ar_listx` file as possible -- these should be handled by the rules file.
- [ ] Improve the overall sound quality of the Arabic voice.
- [ ] Provide support for the MBROLA Arabic voices.

**Comment by ValdisVitolins** [comment]
Recorded consonants in current set have very different voice tunes (fundamental frequencies). These should be remade:
1) better as recording of silent consonants mixed with formant voice on top, because formant voice changes voice tune according to the intonation and rules.
2) by adjusting pitch to more or less monotone pitch of the existing sample files (e.g. with Audacity or similar tool).

**Comment by rhdunn** [comment]
According to comments on the espeak-ng mailing list, the Arabic in espeak-ng is `arq` (Algerian Arabic), not `ar` (Standard Arabic). As such, both dialects/macrolanguages should be supported once the pronunciation rules have been improved and cleaned up.

**Comment by linuxscout** [comment]
What is the problem?
The developed support is ar (standard Arabic) and not Algerian, what's new?

**Comment by ValdisVitolins** [comment]
So, do we need to change existing configuration files from `ar` to `arq` and introduce new set of files for `ar`? I suppose we should do this only if there will be enough contributions for both of these dialects.

**Comment by rhdunn** [comment]
The files should be kept as `ar`. When the language rules are properly fleshed out, they should be checked against Standard and Algerian Arabic. Differences should be handled in a similar way to how English dialects are handled. At that point, an `arq` voice (language) file can be added to enable the Algerian dialect.

**Comment by linuxscout** [comment]
How want to work on Algerian dialect?

**Comment by csisc** [comment]
I thank you for dealing with this important issue. The matter is just that the configuration for Modern Standard Arabic was influenced by the Algerian Arabic phonology. I think that we can keep these files as the ones of Algerian Arabic (arq) and we can adjust these files in order to make the ones of Modern Standard Arabic (ar).

**Comment by csisc** [comment]
Please test the following test to verify what I said:
الذَهَبُ مَعدِنٌ نَفِيسٌ يُستَعمَلُ فِي عَدِيدِ الوَظَائِفِ.

**Comment by linuxscout** [comment]
I'm Algerian, but the arabic voice doesn't use Algerian dialect rules, It's based on Arabic standard rules.
I know that the voice quality is not so good, but the goal is to develop arabic support.
If you can help to adjust phonoloy parameter, you are welcome.

**Comment by csisc** [comment]
I thank you for your answer. What I have seen in the source code is that ج, ذ, and ض do not meet to Modern Standard Arabic phonological features. I did not succeed to try the preview of the configuration in eSpeakedit. If you show me how to do that, I will surely adjust the parameters.

**Comment by csisc** [comment]
dZ meaning [dzˤ] is the pronunciation of ج in Algerian...

**Comment by linuxscout** [comment]
Can you give me the three letters features and parameters for arabic

**Comment by csisc** [comment]
Of course, ج is pronounced as [ʒ]. You can listen to it in https://en.wikipedia.org/wiki/File:Voiced_palato-alveolar_sibilant.ogg.

**Comment by csisc** [comment]
ذ is pronounced as [ð]. You can listen to it in https://en.wikipedia.org/wiki/File:Voiced_dental_fricative.ogg.

**Comment by csisc** [comment]
ض is pronounced as [dˤ].

**Comment by csisc** [comment]
ظ is pronounced as [ðˤ].

**Comment by linuxscout** [comment]
> Of course, ج is pronounced as [ʒ]. You can listen to it in https://en.wikipedia.org/wiki/File:Voiced_palato-alveolar_sibilant.ogg.

You can view that  [dzˤ] is the standard pronunciation, cf. [https://en.wikipedia.org/wiki/Arabic_phonology](https://en.wikipedia.org/wiki/Arabic_phonology) or [https://en.wikipedia.org/wiki/Voiced_palato-alveolar_affricate](https://en.wikipedia.org/wiki/Voiced_palato-alveolar_affricate)
but the [ʒ], is used in dialect, [https://en.wikipedia.org/wiki/Arabic_phonology#endnote_6](https://en.wikipedia.org/wiki/Arabic_phonology#endnote_6).

you can read 

> ^ The phoneme represented by the Arabic letter ǧīm (ج) has many standard pronunciations: [d͡ʒ] in most of the Arabian Peninsula and as the predominant pronunciation of Literary Arabic, [ɡ] in most of Egypt and some regions in Yemen and Oman. This is also a characteristic of colloquial Egyptian and southern Yemeni dialects.[16] In Morocco and western Algeria, it is pronounced as [ɡ] in some words, especially colloquially. In most north Africa and most of the Levant, the standard is pronounced [ʒ], and in certain regions of the Persian Gulf with [j]. In some Sudanese and Yemeni dialects, it may be either [ɡʲ] or [ɟ] as it used to be in Classical Arabic. Foreign words containing /ɡ/ may be transcribed with ج‎, غ‎, ك‎, ق‎, گ‎, ݣ‎ or ڨ‎, mainly depending on the regional spoken variety of Arabic or the commonly diacriticized Arabic letter . Also, /d͡ʒ~ʒ/ (چ only in Egypt) can be used in loanwords where it isn't the standard pronunciation for the letter ǧīm (ج).

For other phonemes, I will check it.

**Comment by csisc** [comment]
In the description of Sebawayh, [d͡ʒ] is the pronunciation of ج. However, the pronunciation of Arabic had changed. When the linguists have decided to create the Modern Standard Arabic that will replace the Classical Arabic by the 19th Century, they have studied the phonology of the Arabic dialects and replaced Classical pronunciation by the mostly used dialectal arabic one. For example, ج became pronounced as [ʒ]. You can verify this by seeing the main references about this issue that are used by specialists in Computational Linguistics. For example, Al-Aswat al-lughawiyah of Ibrahim Anis and Al-Aswat al-lughawiyah of Muhammad Alkhouli...

**Comment by csisc** [comment]
I ask if you can send me the audio file for this input: 
ذَهَبَ الظَبيُ وَ بَقِيَت الضِفدَعَة فِي المُستَنقَعِ ثُمَّ خَرَجَت مِنهُ
I can verify if other matters exist within the configuration.

**Comment by ValdisVitolins** [comment]
Just note that phoneme names from `phsource/phonemes` file would be:
ʒ as `Z`
ð as `D` or `T`

**Comment by csisc** [comment]
I thank Vrdhn0 for his absolutely useful interface available in http://vardhan.net:3000/. I used it to evaluate the eSpeakNG Arabic support. After my discussion with @linuxscout, I have tried to find a paper that explains Standard Arabic phonology. I found the official International Phonetic Association description written in 1990 in https://www.cambridge.org/core/journals/journal-of-the-international-phonetic-association/article/div-classtitlearabicdiv/D54098F7BBE7821861A80617B872FA24. 

- I found that you are right concerning the pronunciation of ج. In fact, it is pronounced as [d͡ʒ].
- ض is pronounced as [dˤ].
- The layout does not support Shaddah
- ع is pronounced as [ʕ]

**Comment by csisc** [comment]
Concerning short vowels, they are not centralized as [e] and [o] according to the eSpeak Arabic support. We can fix this later.

**Comment by csisc** [comment]
The Arabic support as written needs improving. Specifically:
- [ ] Improve the WAV files for [dˤ] and [ʕ].
- [ ] Add conversion rules for the Shaddah
After doing these two adjustments, I will assess the updated output.

**Comment by valdisvi** [comment]
I made [some fixes](https://github.com/valdisvi/espeak-ng/commit/4c5499ec04a3b2698a41cb994dab1192a30f386b) for  [dˤ] and [ʕ]. You can check them [online](http://odo.lv/Espeak).
